packages/notifications/push/amplify_push_notifications/test/amplify_push_noitfications_impl_test.dart

```diff
@@ -113,14 +118,83 @@ void main() {
         authProviderRepo: authProviderRepo,
         config: config,
       );
-      verify(mockServiceProviderClient.registerDevice('123')).called(2);
+      verify(mockServiceProviderClient.registerDevice('123')).called(1);
 
       void tokenHandler(String token) {
         expect(token, '123');
       }
 
       plugin.onTokenReceived.listen(tokenHandler);
     });
+
+    test('should register background processor on Android', () async {
+      await overrideOperatingSystem(
+        const OperatingSystem('android', ''),
+        () async {
+          when(mockPushNotificationsHostApi.getLaunchNotification()).thenAnswer(
+            (_) async => standardAndroidPushMessage,
+          );
+          plugin = TestAmplifyPushNotifications(
+            serviceProviderClient: mockServiceProviderClient,
+            backgroundProcessor: testBackgroundProcessor,
+            dependencyManager: dependencyManager,
+          );
+          await plugin.configure(
+            authProviderRepo: authProviderRepo,
+            config: config,
+          );
+
+          verify(mockPushNotificationsHostApi.registerCallbackFunction(any))
+              .called(1);
+        },
+      );
+    });
+
+    test('should invoke registered event listeners', () async {
+      when(mockPushNotificationsHostApi.getLaunchNotification()).thenAnswer(
+        (_) async => null,
+      );
+
+      plugin = TestAmplifyPushNotifications(
+        serviceProviderClient: mockServiceProviderClient,
+        backgroundProcessor: testBackgroundProcessor,
+        dependencyManager: dependencyManager,
+      );
+      await plugin.configure(
+        authProviderRepo: authProviderRepo,
+        config: config,
+      );
+
+      await testWidgetsFlutterBinding.defaultBinaryMessenger
+          .handlePlatformMessage(
+        notificationOpenedEventChannel.name,
+        notificationOpenedEventChannel.codec
+            .encodeSuccessEnvelope(standardAndroidPushMessage),
+        (_) {},
+      );
+
+      verify(
+        mockServiceProviderClient.recordNotificationEvent(
+          eventType: PinpointEventType.notificationOpened,
+          notification: anyNamed('notification'),
```

Comment:
May be worth to restrictedly verify which message triggers the call of the listener?

Reply:
The content of the notification is not that important to this test plus the `PushNotificationMessage` does not have a unique id to rely the equality on. I am checking for the eventType as it's more important.